registry_module.go

```diff
@@ -94,6 +103,41 @@ type RegistryModuleVersion struct {
 
 	// Relations
 	RegistryModule *RegistryModule `jsonapi:"relation,registry-module"`
+
+	// Links
+	Links map[string]interface{} `jsonapi:"links,omitempty"`
+}
+
+// Upload uploads Terraform configuration files for the provided registry module version. It
+// requires a path to the configuration files on disk, which will be packaged by
+// hashicorp/go-slug before being uploaded.
+func (r *registryModules) Upload(ctx context.Context, rmv RegistryModuleVersion, path string) error {
+	uploadURL, ok := rmv.Links["upload"].(string)
+	if !ok {
+		return fmt.Errorf("Provided RegistryModuleVersion does not contain an upload link")
+	}
+
+	file, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+	if !file.Mode().IsDir() {
+		return ErrMissingDirectory
+	}
+
+	body := bytes.NewBuffer(nil)
+
+	_, err = slug.Pack(path, body, true)
```

Comment:
I think `slug.Pack` here will respect the settings in a `.terraformignore` file if the user has one, which seems like it could possibly lead to this having confusing behavior. Maybe that's actually the more intuitive behavior, though! Either way, I wanted to call it out in case it's something you hadn't considered.

Reply:
Ahh good call out! No, I hadn't thought of that at all.

I'm not sure if it should respect the ignore file or not (my gut reaction is 'it should' but honestly I dunno), but regardless in the name of consistency I'll leave it as is - this is identical to [how configuration versions currently work](https://github.com/hashicorp/go-tfe/blob/db513c343c1f3e84c44895fb27a88df11aab490a/configuration_version.go#L196-L199).

Reply:
That makes sense!